engine/src/main/scala/cromwell/engine/workflow/workflowstore/SqlWorkflowStore.scala

```diff
@@ -53,9 +45,13 @@ case class SqlWorkflowStore(sqlDatabase: WorkflowStoreSqlDatabase) extends Workf
     * Retrieves up to n workflows which have not already been pulled into the engine and sets their pickedUp
     * flag to true
     */
-  override def fetchRunnableWorkflows(n: Int, state: StartableState)(implicit ec: ExecutionContext): Future[List[WorkflowToStart]] = {
-    sqlDatabase.queryWorkflowStoreEntries(n, state.toString, state.afterFetchedState.toString) map {
-      _.toList map fromWorkflowStoreEntry
+  override def fetchRunnableWorkflows(n: Int)(implicit ec: ExecutionContext): Future[List[WorkflowToStart]] = {
+    import cats.instances.list._
+    import cats.syntax.traverse._
+    import common.validation.Validation._
+    sqlDatabase.fetchRunnableWorkflows(n) map {
+      // .get on purpose here to fail the future if something went wrong
+      _.toList.traverse(fromWorkflowStoreEntry).toTry.get
```

Comment:
TIL that's a thing. Is it really not possible to do it nicelier though?

Reply:
I could `flatMap` it and use `Future.fromTry` which would remove the `get` but I don't know how much nicer that would be ?

Reply:
I meant I didn't realise you could throw in a `Future`'s `map` and everything would be fine... I guess it does make sense when I think about it